src/mocompiler.js

```diff
@@ -16,37 +16,42 @@ export default function (table) {
 }
 
 /**
- * Creates a MO compiler object.
- *
- * @constructor
- * @param {import('./index.d.ts').GetTextTranslations} table Translation table as defined in the README
- * @return {import('./index.d.ts').Compiler} Compiler
+ * Prepare the header object to be compatible with MO compiler
+ * @param {{[key: string]: string}} headers the headers
+ * @return {{[key: string]: string}} The prepared header
  */
-function Compiler (table = {}) {
-  this._table = table;
-
-  let { headers = {}, translations = {} } = this._table;
-
-  headers = Object.keys(headers).reduce((result, key) => {
+function prepareMoHeaders (headers) {
+  return Object.keys(headers).reduce((/** @type {{[key: string]: string}} */ result, key) => {
```

Comment:
```suggestion
  return Object.keys(headers).reduce((/** @type {Record<string, string>} */ result, key) => {
```
I would use the `Record` type.